### ISSUE TITLE
Yolo Seg. Live Stock Counter Accuracy with clustered image.

### ISSUE THREAD

**Issue by varaprasad1016** [ultralytics/ultralytics#24424]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Continuing on @glenn-jocher reply as below: For clustered livestock, the main bottleneck is usually missed detections in crowded/overlapping scenes rather than the tracker, so I’d retrain with many more tightly labeled crowded examples, use higher imgsz, and prefer a segmentation model such as yolo26m-seg.pt or yolo26l-seg.pt on your Orin; since this issue is stale, please open a new issue with a few sample frames, your data.yaml, current imgsz/conf/iou settings, and the output of yolo checks so we can give targeted recommendations.[+] Original request: https://github.com/ultralytics/ultralytics/issues/6650

**Comment by UltralyticsAssistant** [comment]
👋 Hello @varaprasad1016, thank you for your interest in Ultralytics 🚀! This is an automated response to help gather the details needed for review, and an Ultralytics engineer will also assist soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
For this livestock counting use case, please share the same key details requested in your note so the team can review it efficiently 🐄:
- A few representative sample frames, especially crowded/overlapping scenes
- Your `data.yaml`
- Your current `imgsz`, `conf`, and `iou` settings
- The output of `yolo checks`
- Any training logs, validation results, and example predictions/tracking outputs
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by varaprasad1016** [comment]
Yolo checks:
_Ultralytics 8.4.15 Python-3.8.10 torch-2.1.0a0+41361538.nv23.06 CUDA:0 (Orin, 62802MiB)
Setup complete (12 CPUs, 61.3 GB RAM, 50.7/56.6 GB disk)

OS Linux-5.10.120-tegra-aarch64-with-glibc2.29
Environment Docker
Python 3.8.10
Install pip
RAM 61.33 GB
Disk 50.7/56.6 GB
CPU ARMv8 Processor rev 1 (v8l) (12 cores)
GPU Orin, 62802 MiB
CUDA 11.4

numpy 1.24.4 ≥ 1.23.0
matplotlib 3.7.5 ≥ 3.3.0
opencv-python (not installed) ≥ 4.6.0
pillow 10.4.0 ≥ 7.1.2
pyyaml 6.0.3 ≥ 5.3.1
requests 2.32.4 ≥ 2.23.0
scipy 1.10.1 ≥ 1.4.1
torch 2.1.0a0 ≥ 1.8.0
torchvision (not installed) ≥ 0.9.0
psutil 7.2.2 ≥ 5.8.0
polars (not installed) ≥ 0.20.0
ultralytics-thop 2.0.18
I

<img width="1412" height="464" alt="Image" src="https://github.com/user-attachments/assets/f01dfe46-331e-46b1-86d6-520b79381a23" />

------------------------------------------------------------------------------------------------------------------------------------------- task: segment mode: train model: yolov8n-seg.pt data: /app/data.yaml
imgsz: 960 
batch: 4 
epochs: 34, val 
mAP50 ≈ 99% by ep 32 
optimizer: auto 
selected AdamW 
amp: true 
deterministic: true 
seed: 0
augmentation hsv_h: 0.015 hsv_s: 0.7 hsv_v: 0.4
fliplr: 0.5 
flipud: 0.0 
degrees: 0.0
translate: 0.1 
scale: 0.5 
perspective: 0.0
mosaic: 1.0 
mixup: 0.0 
cutmix: 0.0 c
opy_paste: 0.0
auto_augment: randaugment erasing: 0.4 close_mosaic: 10

**Comment by rehan243** [comment]
fwiw, we had a super similar headache counting sheep in dense pens — YOLO kept missing the ones smooshed together. The thing is, bumping imgsz helped a bit (we went up to 1280) but honestly, labeling more crowded scenes made way more impact. Tracker tweaks didn't move the needle for us either, just as Glenn said. We ran yolo26l-seg.pt on an Orin NX, but had to crank conf down to 0.15 and play with iou to catch partial masks on overlaps. Also, augmenting the training set with synthetic clusters (using albumentations) made the segmentation pick up edge cases way better, especially on blurry night shots.

Not gonna lie, the data.yaml setup ended up being more important than we expected — we had to split classes by animal age since calves got lost in the cluster. The docs don't mention this, but if you log the output from yolo checks, you'll spot if your imgsz is actually being respected by the pipeline (sometimes it silently falls back to 640, lol). Just sharing since we wasted hours chasing phantom tracker bugs when the real issue was the model missing overlapped masks.

**Comment by varaprasad1016** [comment]
Right!!! Thank you for sharing the findings Rehan, that actually helps, as I did observe that the edge cases were reduced initially as I was changing the imgsz. It makes sense for me to change the model and retrain on the clustered images too. Out of curiosity basing on the Orin set up....did you find yolo26l-seg.pt to be the best performing model yet ?

**Comment by glenn-jocher** [comment]
I haven’t seen one universal “best” on Orin; if crowded-scene recall matters more than FPS, `yolo26l-seg.pt` is usually the first model I’d test, but `yolo26m-seg.pt` often ends up the better production balance, so the right choice is whichever gives you acceptable latency on your exact clips after tuning `imgsz`, `conf`, and `iou` with the [instance segmentation guide](https://docs.ultralytics.com/guides/instance-segmentation-and-tracking/).
